hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/HazelcastWriters.java

```diff
@@ -152,6 +159,45 @@ public static ProcessorMetaSupplier writeListSupplier(@Nonnull String name, @Nul
         ));
     }
 
+    public static ProcessorMetaSupplier writeObservableSupplier(@Nonnull String name) {
+        return new ProcessorMetaSupplier() {
+
+            private JetInstance jet;
+
+            @Override
+            public void init(@Nonnull Context context) {
+                this.jet = context.jetInstance();
+            }
+
+            @Override
+            public void close(@Nullable Throwable error) {
+                ObservableRepository.completeObservable(name, error, jet, System::currentTimeMillis);
+                //todo: force single event?
+            }
+
+            @Nonnull @Override
+            public Function<? super Address, ? extends ProcessorSupplier> get(@Nonnull List<Address> addresses) {
+                return (FunctionEx<? super Address, ? extends ProcessorSupplier>) (Address x) -> new WriterSupplier<>(
```

Comment:
Shouldn't we do a stand-alone ITopic sink and use it here? There will be nothing special once the code from `close` is moved.